### PR TITLE
[14.0][ADD] mrp_subcontracting_resupply_link: New module.

### PR DESCRIPTION
**Pull request by victoralmau**
New module

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT38582

### CODE REVIEW

File: mrp_subcontracting_resupply_link/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Mrp subcontracting resupply link",
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/__manifest__.py]: Links between subcontracting PO and resupply picking

File: mrp_subcontracting_resupply_link/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import SUPERUSER_ID, _, api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    mrp_resupply_picking_count = fields.Integer(
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/models/purchase_order.py]: Change all for being more exact, and at the same time, more concise:

```suggestion
    subcontracting_resupply_count = fields.Integer(
```

File: mrp_subcontracting_resupply_link/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        compute="_compute_mrp_resupply_picking_ids", string="Resupply count", store=True
+    )
+    mrp_resupply_picking_ids = fields.Many2many(
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/models/purchase_order.py]:
```suggestion
    subcontracting_resupply_ids = fields.Many2many(
```

File: mrp_subcontracting_resupply_link/models/stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    mrp_subcontracting_purchase_order_id = fields.Many2one(
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/models/stock_picking.py]:
```suggestion
    subcontracting_purchase_order_id = fields.Many2one(
```

File: mrp_subcontracting_resupply_link/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+This module adds the relationship between the purchase order and resupply and from resupply to the purchase order.
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/readme/DESCRIPTION.rst]:
```suggestion
This module adds convenient links between a purchase order with subcontracting products
and their raw materials resupply pickings.
```

File: mrp_subcontracting_resupply_link/readme/USAGE.rst [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#. Go to *Manufacturing > Products > Products* and create a new subcontracted product with some Vendor (Purchase tab).
+#. Go to *Manufacturing > Products > Products* and create a new one with "Resupply Subcontractor on Order" route checked. [Component]
+#. Go to *Manufacturing > Products > Bill of materials* and create a new one with type = "Subcontracting" and set the subcontractor and component previously created.
+#. Go to *Purchase > Orders > Requests for Quotation* and create a new one to the subcontractor and add subcontracted product.
+#. Click on "Confirm order" button.
+#. A message linked to resupply(s) will have been added to purchase order.
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/readme/USAGE.rst]:
```suggestion
#. A message with the link to the resupply(ies) is added in the purchase order.
```

File: mrp_subcontracting_resupply_link/readme/USAGE.rst [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+#. A "Resupply(s)" smart-button will also be displayed in the purchase order.
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/readme/USAGE.rst]:
```suggestion
#. A "Resupply(ies)" smart-button is also displayed in the purchase order.
#. Clicking on it, it navigates to the linked pickings.
```

File: mrp_subcontracting_resupply_link/readme/USAGE.rst [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+#. A message linked to supplying raw material will have been added in the resupply.
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/readme/USAGE.rst]:
```suggestion
#. A message with the link to the PO is added in the resupply picking.
```

File: mrp_subcontracting_resupply_link/views/purchase_order_view.xml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+                    type="object"
+                    name="action_view_mrp_resupply_picking"
+                    class="oe_stat_button"
+                    icon="fa-truck"
+                    attrs="{'invisible':[('mrp_resupply_picking_ids','=',[])]}"
+                >
+                    <field
+                        name="mrp_resupply_picking_count"
+                        widget="statinfo"
+                        string="Resupply(s)"
```
Comment (pedrobaeza) [inline mrp_subcontracting_resupply_link/views/purchase_order_view.xml]:
```suggestion
                        string="Resupply(ies)"
```

**Review by pedrobaeza — COMMENTED** [review]
Can you increase a bit the coverage?

File: mrp_subcontracting_resupply_link/README.rst [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+Usage
+=====
+
+#. Go to *Manufacturing > Products > Products* and create a new subcontracted product with some Vendor (Purchase tab).
+#. Go to *Manufacturing > Products > Products* and create a new one with "Resupply Subcontractor on Order" route checked. [Component]
+#. Go to *Manufacturing > Products > Bill of materials* and create a new one with type = "Subcontracting" and set the subcontractor and component previously created.
+#. Go to *Purchase > Orders > Requests for Quotation* and create a new one to the subcontractor and add subcontracted product.
+#. Click on "Confirm order" button.
+#. A message linked to resupply(s) will have been added to purchase order.
```
Comment (chienandalu) [inline mrp_subcontracting_resupply_link/README.rst]: It doesn't. I guees this is the issue you were trying to figure out the other day, isn't it?

**Comment by victoralmau** [comment]
Changes done and improved testing to test self-created messages and increase coverage.

File: mrp_subcontracting_resupply_link/README.rst (hunk 10 above)
Comment (victoralmau) [reply mrp_subcontracting_resupply_link/README.rst]: This works correctly, what problem do you have with it?

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Functional review :+1:

**Pull request merged**